python/lsst/donut/viz/aggregate_visit.py

```
@@ -224,26 +231,60 @@ def runQuantum(
         else:
             pairs = self.pairer.run(visitInfoDict)
 
+        # Make dictionaries to match visits and detectors
+        donutRefDict = {
+            (ref.dataId["visit"], ref.dataId["detector"]): ref
+            for ref in inputRefs.donutTables
+        }
+        qualityRefDict = {
+            (ref.dataId["visit"], ref.dataId["detector"]): ref
+            for ref in inputRefs.qualityTables
+        }
+
+        # Find common (visit, detector) pairs
```

Should we raise an error if one or the other is missing here?

---

Sorry, can you elaborate what exactly you mean here?

---

I went ahead and added this check and am going to merge this PR since I will be offline for the rest of the afternoon. Let me know if you want any other checks added, and I can open another PR! 😄

---

Sounds good.  I was just wondering under what conditions we'd expect donutTables without donutQualityTables or vice-versa.